AutoCollection/AsyncHooksScopeManager.ts

```diff
@@ -0,0 +1,62 @@
+import { CorrelationContextManager, CorrelationContext } from "./CorrelationContextManager"
+import { ISpanContext } from "diagnostic-channel";
+import { EventEmitter } from "events";
+
+export interface Span {
```

Comment:
A clone of Span interface so that we don't need to take a Typescript 3 dependency